### PR TITLE
Feat/129 focus sessions repo

### LINKED ISSUE #129: Focus Session Repository
Implement focus session  repository for handling Focus session graphs.

### PR DESCRIPTION
**Pull request by thisshrek**
## What?
Added the `FocusSessionsRepositoryFirestore` class - a complete Firestore implementation of the `FocusSessionsRepository` interface. This includes full CRUD operations and security enforcement for focus sessions operations. Also includes a comprehensive test suite with 11 tests covering all functionality and edge cases. Closes #129.

## Why?
This completes the repository layer for the focus sessions feature, providing the actual persistence layer that interacts with Firebase Firestore. The implementation enforces all business rules and security constraints at the repository level, ensuring data integrity regardless of how the repository is called.

## How?
**Implementation (`FocusSessionsRepositoryFirestore`):**
- Uses Firebase Firestore with a `focusSessions` collection, along with dedicated rules
- Implements all 7 interface methods with proper async/await patterns
- Enforces security rules inline: checks whether the current user's is the focus session's creator before operations.
- Helper methods `snapshotToFocusSession()` and `focusSessionToMap()` for Firestore serialization/deserialization

**Testing (`FocusSessionsRepositoryFirestoreTest` + `FirestoreFocusSessionsGatherlyTest`):**
- 11 comprehensive tests covering CRUD operations, security enforcement and edge cases
- Uses Firebase Emulator Suite with fake JWT tokens for multi-user scenarios
- Tests 3 different users to verify permissions
- Base class `FirestoreFocusSessionsGatherlyTest` handles emulator setup/teardown and user authentication

## Testing?
Full test coverage included with emulator-based integration tests, covering:
- Basic CRUD operations
- Security: creator-only edit/delete
- Edge cases

## Screenshots
image
All 11 tests of the suite pass locally.

## Anything Else?
The implementation of the focus sessions repository was initially adapted from that of the groups repository, thus it is also adapted from the bootcamp.

The `signInWithToken()` helper method in `FirestoreFocusSessionsGatherlyTest` is currently duplicated across multiple test auxiliary classes. This could be refactored into a shared test utility file to reduce code duplication.

### CODE REVIEW

**Review by AACioffi — DISMISSED** [review]
Excellent job on implementing the Focus Sessions repository.

I have just a couple of remarks, but the PR looks good to me otherwise. Approved!

File: app/src/androidTest/java/com/android/gatherly/model/focusSession/FocusSessionsRepositoryFirestoreTest.kt [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    assertEquals(session1.focusSessionId, retrieved.focusSessionId)
+    assertEquals(user1Id, retrieved.creatorId)
+    assertEquals(session1.duration, retrieved.duration)
+    assertEquals(session1.linkedTodoId, retrieved.linkedTodoId)
+    assertEquals(session1.focusSessionId, retrieved.focusSessionId)
+    assertEquals(session1.startedAt, retrieved.startedAt)
+    assertEquals(session1.endedAt, retrieved.endedAt)
+  }
+
+  /** Verifies that getFocusSession() throws a NoSuchElementException when session not found. */
```
[the remark this answers is not in the document]
Comment (thisshrek) [reply app/src/androidTest/java/com/android/gatherly/model/focusSession/FocusSessionsRepositoryFirestoreTest.kt]: Yes, thanks for pointing that out. I was thinking of doing the same as we suggested in the groups repository PR but then kept it this way since it was working.

**Review by AACioffi — APPROVED** [review]
LGTM
